dll/win32/shell32/CDefaultContextMenu.cpp

```diff
@@ -24,30 +39,32 @@ typedef struct _StaticShellEntry_
     HKEY hkClass;
 } StaticShellEntry, *PStaticShellEntry;
 
+#define DFM_FCIDM_SHVIEW_OFFSET 0x7000 // Offset from the menu ids in the menu resource to FCIDM_SHVIEW_*
 
 //
 // verbs for InvokeCommandInfo
 //
-struct _StaticInvokeCommandMap_
+static const struct _StaticInvokeCommandMap_
 {
     LPCSTR szStringVerb;
-    UINT IntVerb;
+    WORD IntVerb;
+    SHORT DfmCmd;
 } g_StaticInvokeCmdMap[] =
 {
     { "RunAs", 0 },  // Unimplemented
     { "Print", 0 },  // Unimplemented
     { "Preview", 0 }, // Unimplemented
-    { "Open", FCIDM_SHVIEW_OPEN },
-    { CMDSTR_NEWFOLDERA, FCIDM_SHVIEW_NEWFOLDER },
-    { "cut", FCIDM_SHVIEW_CUT},
-    { "copy", FCIDM_SHVIEW_COPY},
-    { "paste", FCIDM_SHVIEW_INSERT},
-    { "link", FCIDM_SHVIEW_CREATELINK},
-    { "delete", FCIDM_SHVIEW_DELETE},
-    { "properties", FCIDM_SHVIEW_PROPERTIES},
-    { "rename", FCIDM_SHVIEW_RENAME},
-    { "copyto", FCIDM_SHVIEW_COPYTO },
-    { "moveto", FCIDM_SHVIEW_MOVETO },
+    { "Open",            FCIDM_SHVIEW_OPEN },
+    { CMDSTR_NEWFOLDERA, FCIDM_SHVIEW_NEWFOLDER,  (SHORT)DFM_CMD_NEWFOLDER },
+    { "cut",             FCIDM_SHVIEW_CUT,        /* ? */ },
```

Comment:
Shouldn't you place a default value here? ok it's C++ so maybe it initializes it to zero, but still.

Reply:
C++ will initialize to 0.